libwhisper/WhisperHost.cpp

```diff
@@ -66,21 +68,31 @@ void WhisperHost::inject(Envelope const& _m, WhisperPeer* _p)
 		m_expiryQueue.insert(make_pair(_m.expiry(), h));
 	}
 
-	DEV_GUARDED(m_filterLock)
-	{
-		for (auto const& f: m_filters)
-			if (f.second.filter.matches(_m))
-				for (auto& i: m_watches)
-					if (i.second.id == f.first)
-						i.second.changes.push_back(h);
-	}
+	int rating = 1; // rating for local host is based upon: 1. installed watch; 2. proof of work
+
+	//if (bloomfilter.check)
+	//{
+		// rating *= 10;
+		if (_p) // originated externally
+			DEV_GUARDED(m_filterLock)
+			{
+				for (auto const& f: m_filters)
+					if (f.second.filter.matches(_m))
+						for (auto& i: m_watches)
+							if (i.second.id == f.first)
+							{
+								i.second.changes.push_back(h);
+								rating *= 10;
```

Comment:
+= 10
The nth message isn't exponentially more relevant than a previous messages.